### PR TITLE
Migrate Sylo from master to develop

### LINKED ISSUE #79: Migrate Sylo module: https://github.com/cennznet/cennznet/tree/master/crml/sylo
- Copy from cennznet/master, add into cennznet/develop
- Update plug-blockchain dependencies to point to Plug release candidate 1
- Normalize import names
- Add to runtime definition

### PR DESCRIPTION
**Pull request by aliXsed**
Closes #79

### CODE REVIEW

**Review by jordy25519 — APPROVED** [review]
Nice one. LGTM assuming CI passes

File: crml/sylo/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+[package]
+name = "crml-sylo"
+version = "0.1.0"
```
Comment (cowboy-bebug) [inline crml/sylo/Cargo.toml]: Could u bump up the version to `version = "1.0.0"` <- this is what I have been using for all the modules 😄

File: crml/sylo/src/e2ee.rs [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+	trait Store for Module<T: Trait> as SyloE2EE {
+		PreKeyBundles get(pkbs): map (T::AccountId, DeviceId) => Vec<PreKeyBundle>;
+	}
+}
+
+impl<T: Trait> Module<T> {}
+
+#[cfg(test)]
+pub(super) mod tests {
+	use runtime_io::with_externalities;
```
Comment (cowboy-bebug) [inline crml/sylo/src/e2ee.rs]: This line is no longer required as we will use `execute_with()` instead.
(note: `with_externalities()` is moved to `substrate-externalities` crate)

File: crml/sylo/src/mock.rs [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+	type DispatchVerifier = ();
+}
+
+impl_outer_origin! {
+	pub enum Origin for Test {}
+}
+
+// This function basically just builds a genesis storage key/value store according to
+// our desired mockup.
+pub fn new_test_ext() -> sr_io::TestExternalities<Blake2Hasher> {
```
Comment (cowboy-bebug) [inline crml/sylo/src/mock.rs]: `TestExternalities` doesn't need a type anymore. Try:
```rust
pub fn new_test_ext() -> sr_io::TestExternalities {
```

File: crml/sylo/src/mock.rs [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+	system::GenesisConfig::<Test>::default()
```
Comment (cowboy-bebug) [inline crml/sylo/src/mock.rs]: `system:: GenesisConfig` doesn't need a type anymore. Try:
```rust
	system::GenesisConfig::default()
```

File: crml/sylo/src/e2ee.rs [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+	impl inbox::Trait for Test {}
+	impl response::Trait for Test {}
+	impl groups::Trait for Test {}
+	type E2EE = Module<Test>;
+	type Device = device::Module<Test>;
+	type Response = response::Module<Test>;
+
+	#[test]
+	fn should_add_device() {
+		with_externalities(&mut new_test_ext(), || {
```
Comment (cowboy-bebug) [inline crml/sylo/src/e2ee.rs]: Try:
```rust
fn should_add_device() {
	new_test_ext().execute_with(|| {
```

**Review by cowboy-bebug — CHANGES_REQUESTED** [review]
Nice! Some changes are required 👍

File: crml/sylo/src/e2ee.rs (hunk 5 above)
Comment (aliXsed) [reply crml/sylo/src/e2ee.rs]: Done

File: crml/sylo/src/mock.rs (hunk 4 above)
Comment (aliXsed) [reply crml/sylo/src/mock.rs]: Done

File: crml/sylo/src/mock.rs (hunk 3 above)
Comment (aliXsed) [reply crml/sylo/src/mock.rs]: Done

File: crml/sylo/src/e2ee.rs (hunk 2 above)
Comment (aliXsed) [reply crml/sylo/src/e2ee.rs]: Done

File: crml/sylo/Cargo.toml (hunk 1 above)
Comment (aliXsed) [reply crml/sylo/Cargo.toml]: Done

**Review by cowboy-bebug — COMMENTED** [review]
To make build pass,
- remove all `use runtime_io::with_externalities;`
- replace all `with_externalities()` with `execute_with()`
- replace all `use mock;` ->  `use crate::mock;`

**Review by cowboy-bebug — APPROVED** [review]
Nice work! 🎉
Got some conflicts to resolve below

**Pull request merged**